go/test/endtoend/sharding/verticalsplit/vertical_split_test.go

```diff
@@ -0,0 +1,766 @@
+func checkSrvKeyspaceServedFrom(t *testing.T, cell string, ksname string, expected string, ci cluster.LocalProcessCluster) {
+	srvKeyspace := sharding.GetSrvKeyspace(t, cell, ksname, ci)
+	tabletTypeKeyspaceMap := make(map[string]string)
+	result := ""
+	if srvKeyspace.GetServedFrom() != nil {
+		for _, servedFrom := range srvKeyspace.GetServedFrom() {
+			tabletTy := strings.ToLower(servedFrom.GetTabletType().String())
+			tabletTypeKeyspaceMap[tabletTy] = servedFrom.GetKeyspace()
+		}
+	}
+	if tabletTypeKeyspaceMap["master"] != "" {
+		result = result + fmt.Sprintf("ServedFrom(%s): %s\n", "master", tabletTypeKeyspaceMap["master"])
+	}
+	if tabletTypeKeyspaceMap["rdonly"] != "" {
+		result = result + fmt.Sprintf("ServedFrom(%s): %s\n", "rdonly", tabletTypeKeyspaceMap["rdonly"])
+	}
+	if tabletTypeKeyspaceMap["replica"] != "" {
+		result = result + fmt.Sprintf("ServedFrom(%s): %s\n", "replica", tabletTypeKeyspaceMap["replica"])
+	}
+	assert.Equal(t, expected, result, fmt.Sprintf("Mismatch in srv keyspace for cell %s keyspace %s, expected:\n %s\ngot:\n%s", cell, ksname, expected, result))
+	assert.Equal(t, "", srvKeyspace.GetShardingColumnName(), fmt.Sprintf("Got a sharding_column_name in SrvKeyspace: %s", srvKeyspace.GetShardingColumnName()))
+	assert.Equal(t, "UNSET", srvKeyspace.GetShardingColumnType().String(), fmt.Sprintf("Got a sharding_column_type in SrvKeyspace: %s", srvKeyspace.GetShardingColumnType().String()))
+}
+
+func initializeCluster(t *testing.T) (int, error) {
+	var mysqlProcesses []*exec.Cmd
+	clusterInstance = &cluster.LocalProcessCluster{Cell: cellj, Hostname: hostname}
+
+	// Start topo server
+	if err := clusterInstance.StartTopo(); err != nil {
+		return 1, err
+	}
+
+	for _, keyspaceStr := range []string{sourceKeyspace, destinationKeyspace} {
+		KeyspacePtr := &cluster.Keyspace{Name: keyspaceStr}
+		keyspace := *KeyspacePtr
+		if keyspaceStr == sourceKeyspace {
+			if err := clusterInstance.VtctlProcess.CreateKeyspace(keyspace.Name); err != nil {
+				return 1, err
+			}
+		} else {
+			if err := clusterInstance.VtctlclientProcess.ExecuteCommand("CreateKeyspace", "--served_from", "master:source_keyspace,replica:source_keyspace,rdonly:source_keyspace", "destination_keyspace"); err != nil {
+				return 1, err
+			}
+		}
+		shard := &cluster.Shard{
+			Name: shardName,
+		}
+		for i := 0; i < 4; i++ {
+			// instantiate vttablet object with reserved ports
+			tabletUID := clusterInstance.GetAndReserveTabletUID()
+			tablet := &cluster.Vttablet{
+				TabletUID: tabletUID,
+				HTTPPort:  clusterInstance.GetAndReservePort(),
+				GrpcPort:  clusterInstance.GetAndReservePort(),
+				MySQLPort: clusterInstance.GetAndReservePort(),
+				Alias:     fmt.Sprintf("%s-%010d", clusterInstance.Cell, tabletUID),
+			}
+			if i == 0 { // Make the first one as master
+				tablet.Type = "replica"
+			} else if i == 1 {
+				tablet.Type = "replica"
+			} else {
+				tablet.Type = "rdonly"
+			}
```

Comment:
We can use  vttablet.GetVttabletInstance

Reply:
Done. Refer [PR46](https://github.com/planetscale/vitess/pull/46)